wherehows-web/app/components/datasets/containers/dataset-health.ts

```diff
@@ -30,6 +50,53 @@ export default class DatasetHealthContainer extends Component {
     // Do something in the future
   });
 
+  actions = {
+    /**
+     * Triggered when the user clicks on one of the bars in the summary charts child component, will trigger
+     * a filter for whatever bar they select, unless it already is one in which case we will remove the filter
+     * @param this - Explicit this declaration for typescript
+     * @param filterDatum - Passed in to the action by the child component, contains the tag to be filtered for
+     */
+    onFilterSelect(this: DatasetHealthContainer, filterDatum: IChartDatum): void {
+      const currentFilters = get(this, 'currentFilters');
+      const filterName = filterDatum.name;
+
+      if (currentFilters.has(filterName)) {
+        currentFilters.delete(filterName);
+      } else {
+        // This strange little logic here makes sure we have only one active filter at a time. Taking this away will
```

Comment:
If the business logic is to only have one filter at a time, then it does not seem that a set is the right abstraction here?, why not make the `currentFilters` property of union type `string | null` that points to the filter being used? or if these filters are a static list, a string `enum`? In any case, using a list to account for a single value seems odd...

Reply:
Actually nevermind, found a flaw in this, gonna do a small rewrite.